File: app/src/main/java/com/dhsdevelopments/potato/imagecache/BitmapCacheEntry.java
package com.dhsdevelopments.potato.imagecache;

import android.graphics.Bitmap;

import java.util.ArrayList;
import java.util.List;

class BitmapCacheEntry
{
    Bitmap bitmap;
    boolean loading;
    List<LoadImageCallback> callbacks;

    BitmapCacheEntry( boolean loading ) {
        this.loading = loading;
    }

    public void addCallback( LoadImageCallback callback ) {
        if( callbacks == null ) {
            callbacks = new ArrayList<>();
        }
        callbacks.add( callback );
    }
}

File: app/src/main/java/com/dhsdevelopments/potato/imagecache/BitmapCacheEntry.kt
package com.dhsdevelopments.potato.imagecache

import android.graphics.Bitmap
import java.lang.ref.SoftReference
import java.util.*

internal class BitmapCacheEntry(var loading: Boolean) {
    var bitmap: SoftReference<Bitmap>? = null
    var callbacks: MutableList<LoadImageCallback> = ArrayList()

    fun addCallback(callback: LoadImageCallback) {
        callbacks.add(callback)
    }
}
